src/AppInstallerCLICore/Workflows/UriValidationFlow.cpp

```diff
@@ -0,0 +1,194 @@
+    bool IsBlockedBySmartScreen(Execution::Context& context, const std::string& uri)
+    {
+        auto response = AppInstaller::UriValidation::ValidateUri(uri);
+        switch (response.Decision())
+        {
+        case AppInstaller::UriValidation::UriValidationDecision::Block:
+            AICLI_LOG(Core, Error, << "URI '" << uri << "' was blocked by smart screen. Feedback URL: " << response.Feedback());
+            context.Reporter.Error() << Resource::String::UriBlockedBySmartScreen << std::endl;
+            return true;
+        case AppInstaller::UriValidation::UriValidationDecision::Allow:
+        default:
+            return false;
+        }
+    }
+
+    // Get Uri zone for a given uri or file path.
+    HRESULT GetUriZone(const std::string& uri, Settings::SecurityZoneOptions* zone)
+    {
+        if (!zone)
+        {
+            return E_INVALIDARG;
+        }
+
+        #ifndef AICLI_DISABLE_TEST_HOOKS
+        // For testing purposes, allow the zone to be set via the uri
+        std::smatch match;
+        if (std::regex_search(uri, match, std::regex("/zone(\\d+)/")))
+        {
+            std::string number = match[1].str();
+            *zone = static_cast<Settings::SecurityZoneOptions>(std::stoul(number));
+            return S_OK;
+        }
+        #endif
+
+        DWORD dwZone;
+        auto pInternetSecurityManager = winrt::create_instance<IInternetSecurityManager>(CLSID_InternetSecurityManager, CLSCTX_ALL);
+        auto mapResult = pInternetSecurityManager->MapUrlToZone(AppInstaller::Utility::ConvertToUTF16(uri).c_str(), &dwZone, 0);
+
+        // Ensure MapUrlToZone was successful and the zone value is valid
+        if (FAILED(mapResult))
+        {
+            return mapResult;
+        }
+
+        // Treat all zones higher than untrusted as untrusted
+        if (dwZone > static_cast<DWORD>(Settings::SecurityZoneOptions::UntrustedSites))
+        {
+            *zone = Settings::SecurityZoneOptions::UntrustedSites;
+        }
+        else
+        {
+            *zone = static_cast<Settings::SecurityZoneOptions>(dwZone);
+        }
+
+        return S_OK;
+    }
+
+    // Validate group policy for a given zone.
+    bool IsBlockedByGroupPolicy(Execution::Context& context, const Settings::SecurityZoneOptions zone)
+    {
+        if (!Settings::GroupPolicies().IsEnabled(Settings::TogglePolicy::Policy::AllowedSecurityZones))
+        {
+            AICLI_LOG(Core, Info, << "WindowsPackageManagerAllowedSecurityZones policy is disabled");
+            return false;
+        }
+
+        auto allowedSecurityZones = Settings::GroupPolicies().GetValue<Settings::ValuePolicy::AllowedSecurityZones>();
+        if (!allowedSecurityZones.has_value())
+        {
+            AICLI_LOG(Core, Warning, << "WindowsPackageManagerAllowedSecurityZones policy is not set");
+            return false;
+        }
+
+        if (allowedSecurityZones->find(zone) == allowedSecurityZones->end())

```

Comment:
Nit: Doesn't matter because of how few values there are, but I'd prefer to keep the iterator from `find()` to not have to do a search again with `at()`